src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/PersistFlowUnitAndSummaryTest.java

```diff
@@ -176,35 +176,44 @@ private AppContext createAppContextWithDataNodes(String nodeName, NodeRole role)
     return appContext;
   }
 
+  /**
+   * Add testPersistSummaryOnDataNode() and testPersistSummaryOnMasterNode() into a single UT
+   * This will force both tests to run in sequential and can avoid access contention to the
+   * same db file.
+   * @throws Exception SQL exception
+   */
   @Test
-  public void testPersistSummaryOnDataNode() throws Exception {
-    AppContext appContext = createAppContextWithDataNodes("node1", NodeRole.DATA);
-
-    AnalysisGraph graph = new DataNodeGraph();
+  public void testPersisSummary() throws Exception {
     RcaConf rcaConf = new RcaConf(Paths.get(RcaConsts.TEST_CONFIG_PATH, "rca.conf").toString());
+    RcaConf masterRcaConf = new RcaConf(Paths.get(RcaConsts.TEST_CONFIG_PATH, "rca_elected_master.conf").toString());
     Persistable persistable = PersistenceFactory.create(rcaConf);
+    testPersistSummaryOnDataNode(rcaConf, persistable);
+    testPersistSummaryOnMasterNode(masterRcaConf, persistable);
+    persistable.close();
+  }
+
+  private void testPersistSummaryOnDataNode(RcaConf rcaConf, Persistable persistable) throws Exception {
+    AppContext appContext = createAppContextWithDataNodes("node1", NodeRole.DATA, false);
+
+    AnalysisGraph graph = new DataNodeGraph();
     RCAScheduler scheduler = startScheduler(rcaConf, graph, persistable, this.queryable, appContext);
     // Wait at most 1 minute for the persisted data to show up with the correct contents
     WaitFor.waitFor(() -> {
       String readTableStr = persistable.read();
       System.out.println(readTableStr);
       if (readTableStr != null) {
         return readTableStr.contains("HotResourceSummary") && readTableStr.contains("DummyYoungGenRca")
-                && readTableStr.contains("HotNodeSummary") && readTableStr.contains("HotNodeRcaX")
-                && readTableStr.contains("HighHeapUsageClusterRcaX");
+                && readTableStr.contains("HotNodeSummary") && readTableStr.contains("HotNodeRcaX");
```

Comment:
I see that `HighHeapUsageClusterRcaX` node is created in the `construct` but it is removed from verification here. Can you add a code-comment with the reason ?

Reply:
done